src/customElements/applicationGuidance/Wrapper/editQuickFact.ts

```diff
@@ -0,0 +1,170 @@
+            css`
+                .form {
+                    padding: calc(var(--grid-unit) * 2);
+                }
+
+                input {
+                    background: var(--color-black-alt4);
+                    padding: var(--grid-unit);
+                    border: none;
+                    border-bottom: 1px solid var(--color-black-alt3);
+                    border-radius: var(--border-radius) var(--border-radius) 0 0;
+                    margin-bottom: calc(var(--grid-unit) * 2);
+                    color: var(--color-primary);
+                    font-size: 16px;
+                    width: 100%;
+                    box-sizing: border-box;
+                }
+
+                input:focus {
+                    outline: none;
+                    border-color: var(--color-contrast);
+                }
+
+                label {
+                    font-size: 12px;
+                    line-height: 16px;
+                    color: #6f6f6f;
+                }
+            `,
+        ];
+    }
+
+    firstUpdated() {
+        if (!this.quickFact.title) {
+            this.shadowRoot.querySelector('input')?.focus();
+        }
+    }
+
+    private handleTitleChange = (e: ChangeEvent<HTMLInputElement>) => {
+        const quickFact = this.editQuickFact || this.quickFact;
+        this.editQuickFact = {
+            ...quickFact,
+            title: e.target.value,
+        };
+    };
+
+    private handleBodyChange = (e: CustomEvent) => {
+        const quickFact = this.editQuickFact || this.quickFact;
+        this.editQuickFact = {
+            ...quickFact,
+            bodyMarkdown: e.detail as string,
+        };
+    };
+
+    private handleSave = async () => {
+        if (!this.editQuickFact) {
+            return;
+        }
+
+        this.isSaving = true;
+
+        try {
+            const updatedQuickFact = await this.api.updateQuickFactAsync(
+                this.scope,
+                this.editQuickFact
+            );
+            const event = new QuickFactEvent('quick-fact-updated', updatedQuickFact);
+            this.dispatchEvent(event);
+        } catch (e) {
+            console.error(e);
+        } finally {
+            this.isSaving = false;
+        }
+    };
+
+    private handleCancel = () => {
+        const event = new CustomEvent('cancel');
+        this.dispatchEvent(event);
+    };
+
+    private canSave = () => {
+        return (
+            !this.isSaving &&
+            this.editQuickFact?.title &&
+            this.editQuickFact?.bodyMarkdown &&
+            (this.editQuickFact?.title !== this.quickFact.title ||
+                this.editQuickFact?.bodyMarkdown !== this.quickFact.bodyMarkdown)
+        );
+    };
+
+    renderActions() {
+        return html`
+            <div class="toolbar-header">
+                <button
+                    class="button border"
+                    ?disabled="${this.isSaving}"
+                    @click="${this.handleCancel}"
```

Comment:
*opt*
by using the decorator `@eventOptions({passive:true}:  AddEventListenerOptions)` above the handler you can control the event args.
Also to make a custom event pass through shadow DOM boundaries, you must set both the `composed` and `bubbles` flags to `true`: